src/scripts/utils/utils.js

```diff
@@ -0,0 +1,63 @@
+// tableSort/utils.js
+export function convertToNumber(input) {
+  let s = String(input)
+    .trim()
+    .replace(/[^\d,.\-\s\u00A0\u2007\u202F]/g, '')
+    .replace(/[\s\u00A0\u2007\u202F]/g, '');
+
+  s = s.replace(/,/g, '');
+
+  s = s.replace(/-/g, '');
+
+  if (String(input).includes('-')) {
+    s = '-' + s;
+  }
+
+  if (!s || s === '-' || /^[.]+$/.test(s)) {
+    return NaN;
+  }
+
+  return Number(s);
+}
+
+export function hasInvalidTD(tdA, tdB) {
+  return (
+    !tdA ||
+    !tdB ||
+    tdA.textContent == null ||
+    tdB.textContent == null ||
+    tdA.textContent.trim() === '' ||
+    tdB.textContent.trim() === ''
```

Comment:
Why not just simplify
```suggestion
    !tdA.textContent ||
    !tdB.textContent ||
    !tdA.textContent.trim() ||
    !tdB.textContent.trim()
```